s3/pkg/meta/db/drivers/tidb/bucket.go

```diff
@@ -267,32 +293,28 @@ func (t *TidbClient) ListObjects(ctx context.Context, bucketName, marker, verIdM
 				objectNum[name] = 0
 			}
 			objectNum[name] += 1
-			marker = name
-			//filte row
-			//filte by prefix
-			hasPrefix := strings.HasPrefix(name, prefix)
-			if !hasPrefix {
-				continue
-			}
-			//filte by objectname
+			filter[common.KMarker] = name
+
 			if _, ok := objectMap[name]; !ok {
 				objectMap[name] = struct{}{}
 			} else {
 				continue
 			}
-			//filte by deletemarker
-			if deletemarker {
```

Comment:
if we are NOT interested in rows marked for delete, can we update the query in line 280 to NOT fetch rows marked for delete ?

Reply:
deletemarker is for delete delay, which is not support now, so we do not use it now.